services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/hanke/HankeAttachmentMigrator.kt

```diff
@@ -0,0 +1,93 @@
+package fi.hel.haitaton.hanke.attachment.hanke
+
+import fi.hel.haitaton.hanke.attachment.azure.Container.HANKE_LIITTEET
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContent
+import fi.hel.haitaton.hanke.attachment.common.AttachmentValidator
+import fi.hel.haitaton.hanke.attachment.common.FileClient
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentContentEntity
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentContentRepository
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.MigrationResult
+import fi.hel.haitaton.hanke.attachment.common.UnMigratedHankeAttachment
+import java.util.UUID
+import mu.KotlinLogging
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+
+private val logger = KotlinLogging.logger {}
+
+@Component
+class HankeAttachmentMigrator(
+    private val fileClient: FileClient,
+    private val attachmentRepository: HankeAttachmentRepository,
+    private val contentRepository: HankeAttachmentContentRepository
+) {
+    /** Find an attachment that is not yet migrated, i.e. still has content in the db. */
+    @Transactional(readOnly = true)
+    fun unMigratedAttachment(): UnMigratedHankeAttachment? {
```

Comment:
Unmigrated is one word.

A name that describes what this does would be better. Like `findAttachmentWithDatabaseContent()`.

Reply:
Renamed.